NEXT-12704 - Add store API route (#169)
shopware/docs: guides/plugins/plugins/framework/store-api/add-store-api-route.md

{
    /**
     * @var EntitySearchResult
     */
    protected $object;

    public function __construct(EntitySearchResult $object)
    {
        parent::__construct($object);
    }

    public function getExamples(): ExampleCollection
    {
        /** @var ExampleCollection $collection */
        $collection = $this->object->getEntities();

        return $collection;
    }
}
```
{% endcode %}

## Register route

The last thing we need to do now is to tell Shopware how to look for new routes in our plugin.
This is done with a `routes.xml` file at `<plugin root>/src/Resources/config/` location.
Have a look at the official [Symfony documentation](https://symfony.com/doc/current/routing.html) about routes and how they are registered.

{% code title="<plugin root>/src/Resources/config/routes.xml" %}
```xml
<?xml version="1.0" encoding="UTF-8" ?>
<routes xmlns="http://symfony.com/schema/routing"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://symfony.com/schema/routing
        https://symfony.com/schema/routing/routing-1.0.xsd">
  
    <import resource="../../Core/**/*Route.php" type="annotation" />
</routes>
```
{% endcode %}

> ## Check route in Swagger

Reviewer: I hate Swagger, and most developers do as well 

If I am not implementing the billions of Swagger Annotations because I am lazy, my only chance is the `bin/console debug:router` command, to check, if my routes were loaded

Please add two or three sentences and a reference to it https://symfony.com/doc/current/routing.html#debugging-routes

{
    /**
     * @var EntitySearchResult
     */
    protected $object;

    public function __construct(EntitySearchResult $object)
    {
        parent::__construct($object);
    }

    public function getExamples(): ExampleCollection
    {
        /** @var ExampleCollection $collection */
        $collection = $this->object->getEntities();

        return $collection;
    }
}
```
{% endcode %}

## Register route

The last thing we need to do now is to tell Shopware how to look for new routes in our plugin.
This is done with a `routes.xml` file at `<plugin root>/src/Resources/config/` location.
Have a look at the official [Symfony documentation](https://symfony.com/doc/current/routing.html) about routes and how they are registered.

{% code title="<plugin root>/src/Resources/config/routes.xml" %}
```xml
<?xml version="1.0" encoding="UTF-8" ?>
<routes xmlns="http://symfony.com/schema/routing"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://symfony.com/schema/routing
        https://symfony.com/schema/routing/routing-1.0.xsd">
  
    <import resource="../../Core/**/*Route.php" type="annotation" />
</routes>
```
{% endcode %}

> ## Check the route is working

Reviewer: A route can be registered wrong and not detected by Symfony, whilst being correctly displayed in Swagger. These are completely different topics. 

You do not check for a correct route registration in Swagger.

Please update the documentation to check your routes via the Symfony route debugging tool. To check, if your OpenApi Annotations are correct, you'll have to check Swagger.

Now we are done and can check if our route is working correctly.

### Check route via Symfony debugger

> The first way we can check if everything is working correctly is using the [Symfony route debugger](https://symfony.com/doc/current/routing.html#debugging-routes).

Reviewer: When you rewrite this, use following terminology:

```suggestion
To check, if our route was registered correctly, we can use the [Symfony route debugger](https://symfony.com/doc/current/routing.html#debugging-routes).
```

> ```bash
> $ ./bin/console debug:router store-api.example.search
> ```

Reviewer: ``````suggestion
{% code %}
```bash
$ ./bin/console debug:router store-api.example.search
```
{% endcode %}
``````
